File: server/router/emag/emag_order.go
package emag

import (
	"github.com/gin-gonic/gin"
)

type EmagOrderRouter struct{}

// InitEmagOrderRouter 初始化订单路由
func (e *EmagOrderRouter) InitEmagOrderRouter(Router *gin.RouterGroup) {
	emagOrderRouter := Router.Group("emagOrder")
	{
		emagOrderRouter.GET("getList", emagOrderApi.GetOrderList)          // 分页获取订单列表
		emagOrderRouter.POST("sync", emagOrderApi.SyncOrders)              // 同步订单
		emagOrderRouter.GET("getCountryList", emagOrderApi.GetCountryList) // 获取国家列表
	}
}

File: web/src/api/emagOrder.js
import service from '@/utils/request'

// 分页获取订单列表
export const getOrderList = (params) => {
  return service({
    url: '/emagOrder/getList',
    method: 'get',
    params
  })
}

// 同步订单
export const syncOrders = (data) => {
  return service({
    url: '/emagOrder/sync',
    method: 'post',
    data
  })
}

// 获取国家列表
export const getCountryList = () => {
  return service({
    url: '/emagOrder/getCountryList',
    method: 'get'
  })
}

